lib/tictactoe.rb

```diff
@@ -0,0 +1,66 @@
+class TicTacToe
+  attr_reader :player1, :player2, :board
+  def initialize(player1, player2, board)
+    @player1 = player1
+    @player2 = player2
+    @board = board
+  end
+
+  def player_move
+    # Figures out which player should do their next move
+    puts 'Player makes a move'
+  end
+
+  def print_board()
+    # prints 2D array
+    # puts 'Now, your move is displayed on the board'
+    Gem.win_platform? ? (system 'cls') : (system 'clear')
+    @board.matrix.each_with_index do |row, i|
+      row.each_with_index do |col, j|
+        j < 2 ? print(" #{col} |") : print(" #{col} ")
+      end
+      i < 2 && puts("\n---+---+---")
+    end
+    puts ''
+  end
+
+  def play_game
+    @move_count = 0
+    current_player = nil
+    (1..9).each do
+      current_player = @move_count.even? ? player1 : player2
+      print "#{current_player.name} Enter a row: "
+
+      row = gets.chomp.to_i
+      next unless valid_input?(row)
+
+      print 'Enter a column: '
+      column = gets.chomp.to_i
+      next unless valid_input?(column)
+
+      if board.move_valid?(row - 1, column - 1)
+        board.apply_coords(row - 1, column - 1, current_player.sign)
+        @move_count += 1
+      end
+      print_board
+      puts 'Its a draw' if draw?
```

Comment:
## Game loop shouldn't finish only nine moves

- Please follow this example to take in mind when the game should finish
  - [ ] Edit this loop taking in ming the game in this case the loop finish when there's a winner or it's a draw

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

Reply:
Hi 
We don't understand. How is the game suppose to run if we don't have the logic added? We made the range 1 to 9 because that's how many moves it will take for a draw to be made. We didn't add the win method, since we don't want to add all the logic of the game.